src/test/java/org/opensearch/security/auditlog/integration/BasicAuditlogTest.java

```diff
@@ -551,6 +551,70 @@ public void testIndexPattern() throws Exception {
         Assert.assertTrue(res.getBody().contains("auditlog-20"));
     }
 
+    @Test
+    public void testDataStreamAsBackend() throws Exception {
+
+        // Set config to use a datastream as auditlog.
+        Settings additionalSettings = Settings.builder()
+            .put("plugins.security.audit.type", "internal_opensearch")
+            .put(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_LOG_REQUEST_BODY, false)
+            .put(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_RESOLVE_INDICES, false)
+            .put(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_CONFIG_DISABLED_TRANSPORT_CATEGORIES, "NONE")
+            .put(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_CONFIG_DISABLED_REST_CATEGORIES, "NONE")
+            .put("plugins.security.audit.threadpool.size", 10) // must be greater 0
+            .put("plugins.security.audit.config.index", "auditlog-datatream")
+            .build();
+
+        setup(additionalSettings);
+
+        // Prepare auditlog datastream template
+        var templateBody = "{"
+            + "\"index_patterns\": [ \"auditlog-datatream\" ],"
+            + "\"priority\": 90,"
+            + "\"data_stream\": {"
+              + "\"timestamp_field\": {"
+              + "\"name\": \"@timestamp\""
+            + "}}"
+            + "}";
+
+        HttpResponse res = restHelper().executePostRequest("_index_template/auditlog-datastream", templateBody, encodeBasicHeader("admin", "admin"));
+        Assert.assertEquals(res.getStatusCode(), HttpStatus.SC_OK);
+
+        setupStarfleetIndex();
+
+        HttpResponse res2 = rh.executeGetRequest("_data_stream/auditlog-datatream", encodeBasicHeader("admin", "admin"));
+        Assert.assertEquals(res2.getStatusCode(), HttpStatus.SC_OK);
+
+        HttpResponse res3 = rh.executeGetRequest("auditlog-datatream/_stats", encodeBasicHeader("admin", "admin"));
```

Comment:
Getting the broad stats is useful, but we want to be sure the correct auto log entries are being generated, lets query the datastream's contents and be sure we have the expected audit log entries.

If the specific entries are missing the tests should fail, example testing depth [1]

- [1] https://github.com/opensearch-project/security/blob/main/src/test/java/org/opensearch/security/auditlog/compliance/ComplianceAuditlogTest.java#L309